HW1/P6/P6.py

```diff
@@ -0,0 +1,68 @@
+from pyspark import SparkContext
+import numpy as np
+sc =SparkContext()
+
+nPartitions = 20
+# Read in files
+data = sc.textFile("shakespear.txt").collect()
+
+rdd = sc.parallelize(data, nPartitions).flatMap(lambda x: x.split())
+# Function that filters out words as required.
+def contextFilter(rdd):
+	rdd = rdd.filter(lambda x: x.isdigit() == False)
+	rdd = rdd.filter(lambda x: x.isupper() == False)
+	rdd = rdd.filter(lambda x: x[:-1].isupper() == False)
+	return rdd
+
+rdd = contextFilter(rdd)
+
+# For each word, add its first and second lag.
+def concatenate(rdd):
+	words1 = rdd.collect()
+	words2 = words1[1:] + ["/"]
+	words3 = words2[1:] + ["/"]
+
```

Comment:
I think you can create rdd2 and rdd3 directly from rdd rather than collect it first.